src/tests/uat/recommendation_uat.py

```diff
@@ -0,0 +1,423 @@
+
+    Returns:
+        bool: True if comparison was successful, False if not.
+    """
+    result = compare(context, application.type, raw_data, 'type')
+    result = compare(context, application.value, raw_data, 'value') and result
+    return result
+
+
+def compare_new_rule(context, new_rule, raw_data):
+    """
+    Compare a RecommendationNewRule object against the data that should represent it.
+
+    Args:
+        context (str): The context of the comparison, used for printing error messages.
+        new_rule (RecommendationNewRule): The RecommendationNewRule object to compare.
+        raw_data (dict): The dictionary containing what should be the same data.
+
+    Returns:
+        bool: True if comparison was successful, False if not.
+    """
+    result = compare(context, new_rule.action, raw_data, 'action')
+    result = compare(context, new_rule.certificate_authority, raw_data, 'certificate_authority') and result
+    result = compare(context, new_rule.filename, raw_data, 'filename') and result
+    result = compare(context, new_rule.include_child_processes, raw_data, 'include_child_processes') and result
+    result = compare(context, new_rule.operation, raw_data, 'operation') and result
+    result = compare(context, new_rule.override_list, raw_data, 'override_list') and result
+    result = compare(context, new_rule.override_type, raw_data, 'override_type') and result
+    result = compare(context, new_rule.path, raw_data, 'path') and result
+    result = compare(context, new_rule.sha256_hash, raw_data, 'sha256_hash') and result
+    result = compare(context, new_rule.signed_by, raw_data, 'signed_by') and result
+    if new_rule.application_:
+        result = compare_application(f"{context}.application", new_rule.application_,
+                                     raw_data['application']) and result
+    else:
+        result = check_not_present(context, raw_data, 'application') and result
+    return result
+
+
+def compare_workflow(context, workflow, raw_data):
+    """
+    Compare a RecommendationWorkflow object against the data that should represent it.
+
+    Args:
+        context (str): The context of the comparison, used for printing error messages.
+        workflow (RecommendationWorkflow): The RecommendationWorkflow object to compare.
+        raw_data (dict): The dictionary containing what should be the same data.
+
+    Returns:
+        bool: True if comparison was successful, False if not.
+    """
+    result = compare(context, workflow.changed_by, raw_data, 'changed_by')
+    result = compare(context, workflow.create_time, raw_data, 'create_time') and result
+    result = compare(context, workflow.ref_id, raw_data, 'ref_id') and result
+    result = compare(context, workflow.status, raw_data, 'status') and result
+    result = compare(context, workflow.update_time, raw_data, 'update_time') and result
+    result = compare(context, workflow.comment, raw_data, 'comment') and result
+    return result
```

Comment:
Can you print a message if 'result' is false before returning it?

Reply:
The `compare()` function does that.

```
 if not result:
        print(f"field value {raw_field} did not match in {context} - object value {object_field}, "
              f"raw value {raw_data.get(raw_field, None)}")
```